airflow/models/baseoperator.py

```diff
@@ -1659,6 +1629,115 @@ def defer(
         """
         raise TaskDeferred(trigger=trigger, method_name=method_name, kwargs=kwargs, timeout=timeout)
 
+    def map(self, **kwargs) -> "MappedOperator":
+        return MappedOperator(
+            operator_class=type(self),
+            operator=self,
+            task_id=self.task_id,
+            task_group=getattr(self, 'task_group', None),
+            dag=getattr(self, '_dag', None),
+            start_date=self.start_date,
+            end_date=self.end_date,
+            partial_kwargs=self.__init_kwargs,
+            mapped_kwargs=kwargs,
+        )
+
+
+def _validate_kwarg_names_for_mapping(cls: Type[BaseOperator], func_name: str, value: Dict[str, Any]):
```

Comment:
I wonder if `_validate_arg_names` and this have similar logic to be refactored out.

Reply:
Similar, but the way this gets the possible names (most of the function) is quite differnt, so I couldn't work out a way of really sharing it